packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/sticky-toc.twig

```diff
@@ -8,20 +8,21 @@
   {# Static display on desktop #}
   <div class="ma__sticky-toc__inner">
     <div class="ma__sticky-toc__container">
-      <h2 class="ma__sticky-toc__title" id="page-contents">{{ stickyTOC.title }}</h2>
-      <nav class="ma__sticky-toc__links" aria-labelledby="page-contents">
+      <h2 class="ma__sticky-toc__title">{{ stickyTOC.title }}</h2>
+      <nav class="ma__sticky-toc__links">
         {# Mobile toggle button #}
-        <button class="ma__sticky-toc__toggle-link" aria-haspopup="true" aria-expanded="false"><span class="ma__visually-hidden js-sticky-toc__state">Show</span><span class="ma__visually-hidden"> table of contents</span><span aria-hidden="true">+</span></button>
+        <button class="ma__sticky-toc__toggle-link" aria-haspopup="true" aria-expanded="false"><span class="ma__visually-hidden"> table of contents</span><span aria-hidden="true">+</span></button>
         {# Desktop layover #}
         <div class="ma__sticky-toc__column">
-          <div class="ma__sticky-toc__secondary-label">{{ stickyTOC.title }} <button class="secondary-label-close" aria-expanded="true"><span class="ma__visually-hidden">Hide Table of contents</span><span aria-hidden="true">x</span></button></div>
-
-          {% for link in stickyTOC.links %}
-          <div class="ma__sticky-toc__link" data-link="#{{ link.href }}">
-            <a href="#{{ link.href }}" {{ link.info ? 'title="{{ link.info }}"' : '' }}>{{ link.text }}</a>
-          </div>
-          {% endfor %}
-
+          <div class="ma__sticky-toc__secondary-label"><span aria-hidden="true">{{ stickyTOC.title }}</span> <button class="secondary-label-close" aria-expanded="true"><span class="ma__visually-hidden">Hide Table of contents</span><span aria-hidden="true">x</span></button></div>
+          <ul role="menu">
+            {# The section of the for loop is not used.  List item and link are generated by JS. #}
+            {% for link in stickyTOC.links %}
+              <li class="ma__sticky-toc__link" data-link="#{{ link.href }}" role="none">
```

Comment:
That's a good question.  You see the `ul` has the **menu** role.  The menu has a group of menu items, which are links as a child element of each `li`.  So, `li > a` has `role="menuitem"`.  The `li`s are still list items although their parent `ul` is now **menu**.  To prevent the `li`s to be announced as list items by screen readers, they are given the role `none`.  The semantics is maintained by the roles, `menu` and `menuitem`.  The list structure as `ul` lets screen readers recognize how many items are in the menu and tell that to their users.

Reply:
interesting, thanks for the explanation! good to know!